xbmc/video/windows/GUIWindowVideoNav.cpp

```diff
@@ -268,6 +268,112 @@ std::string CGUIWindowVideoNav::GetQuickpathName(const std::string& strPath) con
   }
 }
 
+int CGUIWindowVideoNav::SelectFirstUnwatchedItem()
+{
+  if (m_vecItems->IsVideoDb())
+  {
+    NODE_TYPE NodeType = CVideoDatabaseDirectory::GetDirectoryChildType(m_vecItems->GetPath());
+
+    if (NodeType == NODE_TYPE_SEASONS || NodeType == NODE_TYPE_EPISODES)
+      return CSettings::Get().GetInt("videolibrary.tvshowselectfirstunwatcheditem");
+  }
+
+  return 0;
+}
+
+int CGUIWindowVideoNav::IncludeAllSeasonsAndSpecials()
+{
+  return CSettings::Get().GetInt("videolibrary.tvshowsincludeallseasonsandspecials");
+}
+
+int CGUIWindowVideoNav::GetFirstUnwatchedItemIndex(bool includeAllSeasons, bool includeSpecials)
+{
+  int iIndex = 0;
+
+  int iUnwatchedSeason = INT_MAX;
+
+  // Run through the list of items and find the season number of the first season with unwatched episodes
+  for (int i = 0; i < m_vecItems->Size(); ++i)
+  {
+    CFileItemPtr pItem = m_vecItems->Get(i);
+    if (pItem->IsParentFolder() || !pItem->HasVideoInfoTag())
+      continue;
+
+    CVideoInfoTag *pTag = pItem->GetVideoInfoTag();
+
+    if ((!includeAllSeasons && pTag->m_iSeason < 0) || (!includeSpecials && pTag->m_iSeason == 0))
+      continue;
+
+    // Is the season unwatched, and is its season number lower than the currently identified
+    // first unwatched season
+    if (pTag->m_playCount == 0 && pTag->m_iSeason < iUnwatchedSeason)
+    {
+      iUnwatchedSeason = pTag->m_iSeason;
+      iIndex = i;
+    }
+  }
+
+  NODE_TYPE NodeType = CVideoDatabaseDirectory::GetDirectoryChildType(m_vecItems->GetPath());
+  if (NodeType == NODE_TYPE::NODE_TYPE_EPISODES)
+  {
+    iIndex = 0;
+
+    int iUnwatchedEpisode = INT_MAX;
+
+    // Now run through the list of items and check episodes from the season identified above
+    // to find the first (lowest episode number) unwatched epsisode.
+    for (int i = 0; i < m_vecItems->Size(); ++i)
+    {
+      CFileItemPtr pItem = m_vecItems->Get(i);
+      if (pItem->IsParentFolder() || !pItem->HasVideoInfoTag())
+        continue;
+
+      CVideoInfoTag *pTag = pItem->GetVideoInfoTag();
+
+      // Does the episode belong to the unwatched season
+      if (pTag->m_iSeason != iUnwatchedSeason)
+        continue;
+
+      // Is the episode unwatched, and is its epsiode number lower than the currently
+      // identified first unwatched episode
+      if (pTag->m_playCount == 0 && pTag->m_iEpisode < iUnwatchedEpisode)
+      {
+        iUnwatchedEpisode = pTag->m_iEpisode;
+        iIndex = i;
+      }
+    }
+  }
+
+  return iIndex;
+}
+
+bool CGUIWindowVideoNav::Update(const std::string &strDirectory, bool updateFilterPath /* = true */)
+{
+  if (!CGUIWindowVideoBase::Update(strDirectory, updateFilterPath))
+    return false;
+
+  // Check if we should select the first unwatched item
+  int iSelectFirstUnwatched = SelectFirstUnwatchedItem();
+  if (iSelectFirstUnwatched > 0)
+  {
+    int iItem = m_viewControl.GetSelectedItem();
```

Comment:
if you change this var to be a bool and rename the var to `bIsItemSelected` you can drop the comment 2 lines below.

```
bool bIsItemSelected = (m_viewControl.GetSelectedItem() > 0);
```